server/service/src/sync/translations/invoice.rs

```diff
@@ -815,31 +883,35 @@ fn legacy_invoice_type(_type: &InvoiceType) -> Option<LegacyTransactType> {
 }
 
 fn legacy_invoice_status(t: &InvoiceType, status: &InvoiceStatus) -> Option<LegacyTransactStatus> {
+    // TODO: Should we return none if it's an invalid status? E.g. Shipped for a Prescription?
     let status = match t {
         InvoiceType::OutboundShipment | InvoiceType::SupplierReturn => match status {
             InvoiceStatus::New => LegacyTransactStatus::Sg,
             InvoiceStatus::Allocated => LegacyTransactStatus::Sg,
             InvoiceStatus::Picked => LegacyTransactStatus::Cn,
             InvoiceStatus::Shipped => LegacyTransactStatus::Fn,
             InvoiceStatus::Delivered => LegacyTransactStatus::Fn,
+            InvoiceStatus::Received => LegacyTransactStatus::Fn,
             InvoiceStatus::Verified => LegacyTransactStatus::Fn,
-            InvoiceStatus::Cancelled => LegacyTransactStatus::Fn, // TODO enable cancelled status to sync with mSupply https://github.com/msupply-foundation/open-msupply/issues/6495
+            InvoiceStatus::Cancelled => LegacyTransactStatus::Fn,
         },
         InvoiceType::InboundShipment | InvoiceType::CustomerReturn => match status {
             InvoiceStatus::New => LegacyTransactStatus::Nw,
             InvoiceStatus::Allocated => LegacyTransactStatus::Nw,
             InvoiceStatus::Picked => LegacyTransactStatus::Nw,
             InvoiceStatus::Shipped => LegacyTransactStatus::Nw,
-            InvoiceStatus::Delivered => LegacyTransactStatus::Cn,
+            InvoiceStatus::Delivered => LegacyTransactStatus::Nw,
+            InvoiceStatus::Received => LegacyTransactStatus::Cn,
             InvoiceStatus::Verified => LegacyTransactStatus::Fn,
-            InvoiceStatus::Cancelled => LegacyTransactStatus::Fn, // TODO enable cancelled status to sync with mSupply https://github.com/msupply-foundation/open-msupply/issues/6495
+            InvoiceStatus::Cancelled => LegacyTransactStatus::Fn,
         },
         InvoiceType::Prescription => match status {
             InvoiceStatus::New => LegacyTransactStatus::Nw,
             InvoiceStatus::Allocated => LegacyTransactStatus::Cn,
             InvoiceStatus::Picked => LegacyTransactStatus::Cn,
             InvoiceStatus::Shipped => LegacyTransactStatus::Fn,
-            InvoiceStatus::Delivered => LegacyTransactStatus::Fn,
+            InvoiceStatus::Delivered => LegacyTransactStatus::Sg,
```

Comment:
Probably should be 'InvoiceStatus::New ' ? I am not sure about this 'Sg' status now